drivers/power/ltc3337/ltc3337.c

```diff
@@ -0,0 +1,731 @@
+
+	reg_value = dev->latched_reg_a;
+
+	//Only the upper 8 bits are utilized
+	data_to_set = (counter_reg_val >> 8);
+
+	//If rounding was requested, check the if any of the lower bits are set,
+	//and if we wont overflow dataToSet beyond 8-bits, increment
+	if((round_up) && (counter_reg_val & 0xFF) && (data_to_set != 0xFF))
+		data_to_set++;
+
+	reg_value &= ~(LTC3337_RA_ALARM_LVL_MSK);
+	reg_value |= no_os_field_prep(LTC3337_RA_ALARM_LVL_MSK, data_to_set);
+
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, reg_value);
+	if(ret)
+		return ret;
+
+	dev->latched_reg_a = reg_value;
+	return 0;
+}
+
+/**
+ * Sets the level for the coulomb counter accumulated charge register.
+ * The input value is the raw register value, and can be calculated using
+ * LTC3337_CalculateChargeRegister or manually by hand.
+ * IMPORTANT: Only the upper 8-bits of the value are utilized. Use the roundUp
+ * flag to have the driver optionally round up to the next valid bit
+ * @param dev - Device to configure
+ * @param reg_value - Register value, as if all 16-bits were utilized
+ * @param round_up - 1 -Have the driver round up to next usable alarm bit
+ * @returns 0 on success, Non-0 on error
+ */
+int ltc3337_set_accumulated_charge(struct ltc3337_dev* dev, uint16_t reg_value,
+				   uint8_t round_up)
+{
+	uint16_t data_to_set;
+
+	//Input parameter NULL check
+	if(!dev)
+		return -EINVAL;
+
+	//Only upper 8-bits (MSB) are writable
+	data_to_set = reg_value & 0xFF00;
+
+	//If rounding up was requested, check if any of the lower 8-bits of the
+	//original value are set, and verify dataToSet will not overflow 16-bits
+	//if we increment
+	//Add 1 to the MSB of the data to set to account for rounding
+	//request
+	if((round_up) && (reg_value & 0xFF) && (data_to_set != 0xFF00))
+		data_to_set += 0x100;
+
+	return ltc3337_write_reg(dev, LTC3337_REG_B, data_to_set);
+}
+
+/**
+ * Reads and clears the interrupt status of the chip.  Since the die temp is
+ * also provided in the interrupt register, temperature is optionally provided
+ * as well. This allows a user to poll interrupt status and not accidentally
+ * clear interrupts by reading temperature inbetween interrupt polls
+ * @param dev - Device to read from
+ * @param int_field - Interrupt status bits
+ * @param temp_c - Location to store temperature (or NULL)
+ * @returns 0 on success, Non-0 on error
+ */
+int ltc3337_get_and_clear_interrupts(struct ltc3337_dev* dev,
+				     uint8_t* int_field,
+				     int16_t* temp_c)
+{
+	uint16_t int_reg_value;
+	uint16_t write_reg_value;
+	int ret;
+
+	//Input parameter NULL checks
+	if((!dev) || (!int_field))
+		return -EINVAL;
+
+	ret = ltc3337_read_reg(dev, LTC3337_REG_C, &int_reg_value);
+	if(ret)
+		return ret;
+
+	write_reg_value = dev->latched_reg_a;
+	write_reg_value |= (LTC3337_RA_CLEAR_INT);
+	ret = ltc3337_write_reg(dev, LTC3337_REG_A, write_reg_value);
+
+	*int_field = 0;
+	*int_field |= (int_reg_value & LTC3337_RC_ADC_READY) ? LTC3337_INT_ADC_READY :
```

Comment:
this looks like a `switch ... case`

Reply:
Not a switch, possible for multiple bits to be set.

Original commit hid the driver details from the user, and mapped interrupt register bits to user facing bits in the header. Now that the register details are located in the header, this was changed to just give them the entire register.